app/routes/login.tsx

```diff
@@ -57,15 +84,20 @@ const Login: FC = () => {
     const firebaseClientAuth = initFirebaseClient(firebaseClientConfig);
 
     const githubAuth = new GithubAuthProvider();
+    githubAuth.addScope("read:user");
 
     const result = await signInWithPopup(firebaseClientAuth, githubAuth);
     const idToken = await result.user.getIdToken();
 
-    if (idToken) {
+    const credential = GithubAuthProvider.credentialFromResult(result);
+    const accessToken = credential?.accessToken;
```

Comment:
Users log in through the `signInWithPopup` method, which opens a popup to GitHub. Once that succeeds, we grabbed the `accessToken` returned by the credential exchange. That `accessToken` will come in handy later, but for now, we submit it as part of the form payload.

Reply:
Praise: I love how simple this is!